plugins/package-managers/go/src/funTest/kotlin/utils/GoSupportFunTest.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) 2023 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.plugins.packagemanagers.go.utils
+
+import io.kotest.assertions.throwables.shouldThrow
+import io.kotest.core.spec.style.WordSpec
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.string.shouldContainIgnoringCase
+
+class GoSupportFunTest : WordSpec({
+    "escapeVersion()" should {
+        "escape uppercase letters from a version string" {
+            val version = "v0.1.0-M4.0.20231102094829-08e0c3cd016c"
+            val escapedVersion = escapeVersion(version)
+            escapedVersion shouldBe "v0.1.0-!m4.0.20231102094829-08e0c3cd016c"
+        }
+
+        "throw exception is exclamation mark is used in version string" {
+            val version = "v1.0.0-finally-it-works!"
+
+            val exception = shouldThrow<IllegalArgumentException> {
+                escapeVersion(version)
+            }
+
+            exception.message shouldContainIgnoringCase "exclamation mark"
```

Comment:
I'd prefer to drop the assertion in line 42 for resilience of the test.

Reply:
Good argument.